### ISSUE TITLE
Calendar, TreeSelect, MultiSelect: Invalid aria-controls attribute value. Overlay missing an id attribute. 

### ISSUE THREAD

**Issue by TaneliTuomola** [primefaces/primeng#14980]
In addition to what was mentioned in this ticket, the Calendar's **panelId** does not refer to any element in the Calender.  I think the Calendar template needs to be updated so that the **div.p-datepicker** element should have an **id** attribute set to **panelId**:
```
...
            <div
                #contentWrapper
           [attr.id]="panelId"
                [class]="panelStyleClass"
                [ngStyle]="panelStyle"
                [ngClass]="{
                    'p-datepicker p-component': true,
                    'p-datepicker-inline': inline,
                    'p-disabled': disabled,
                    'p-datepicker-timeonly': timeOnly,
                    'p-datepicker-multiple-month': this.numberOfMonths > 1,
                    'p-datepicker-monthpicker': view === 'month',
                    'p-datepicker-touch-ui': touchUI
                }"
...
```

_Originally posted by @barbGit2 in https://github.com/primefaces/primeng/issues/14358#issuecomment-1900961686_

Tested with primeNg 17.10.0: https://primeng.org/calendar
